scopes/code/ui/code-compare/code-compare-view/code-compare-view.tsx

```diff
@@ -69,6 +73,35 @@ export function CodeCompareView({
   }, [baseId?.toString(), originalFileContent, modifiedFileContent]);
 
   const monacoRef = useRef<any>();
+  const containerRef = useRef() as React.MutableRefObject<HTMLDivElement>;
+  const lastScrollLeftRef = useRef<number>(0);
+
+  /**
+   * checks if the horizontal scroll position has changed since the last scroll event
+   * if changes, it checks if the user has scrolled by a significant threshold (more than 10 pixels).
```

Comment:
why not just disable the 'back' when horizontal scroll in the code compare block?

Reply:
I think you can fix with only css. please check if you can use `overscroll-behavior-x`